.github/workflows/release.yml

```diff
@@ -0,0 +1,49 @@
+name: Release
+
+on:
+  push:
+    branches: [main]
+  workflow_dispatch:
+
+concurrency:
+  group: build-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  release:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-20.04] # TODO: macos-11
+      fail-fast: false
+    env:
+      GH_TOKEN: ${{ github.token }}
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: hendrikmuhs/ccache-action@v1
+        if: matrix.os == 'ubuntu-20.04' || matrix.os == 'macos-11'
+
+      - name: Get short git SHA
+        id: sha
+        run: echo "GIT_SHA_SHORT=$(git rev-parse --short HEAD)" >> $GITHUB_OUTPUT
+
+      - name: Install system dependencies (Ubuntu)
+        run: ./scripts/install_deps_ubuntu.sh
+        if: matrix.os == 'ubuntu-20.04'
+
+      - name: Install system dependencies (Mac)
+        run: ./scripts/install_deps_mac.sh
+        if: matrix.os == 'macos-11'
+
+      - name: Install venv dependencies
+        run: ./bootstrap_venv.sh
+
+      - name: Create and upload release
+        run: |
+          git tag ${{ steps.sha.outputs.GIT_SHA_SHORT }}
+          git push origin --tags
+          tar -czvf venv-${{ matrix.os }}.tar.gz venv
```

Comment:
there is a challenge making the venv relocatable as I think the paths get baked into things.  perhaps this needs to be just the installed libraries and not the venv (user can recreate the venv on their side pretty easily).